sqlglot/transforms.py

```diff
@@ -368,6 +368,16 @@ def _udtf_type(u: exp.Unnest, has_multi_expr: bool) -> t.Type[exp.Func]:
                 expression.args["joins"].remove(join)
 
                 alias_cols = alias.columns if alias else []
+
+                # Handle UNNEST to LATERAL VIEW EXPLODE: Exception is raised when there is 0 or more then 2 aliases for single unnest column.
+                # Spark LATERAL VIEW EXPLODE requires single alias for array/struct and two for Map type column unlike unnest in trino/presto which can take zero or more.
+                # Refs: https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-lateral-view.html
+
+                if not has_multi_expr and len(alias_cols) not in [1, 2]:
+                    raise UnsupportedError(
+                        "Please provide alias to unnest column as per column type to support the transformation."
```

Comment:
Let's change that to 3rd person, something along the lines of:

`CROSS JOIN UNNEST to LATERAL VIEW EXPLODE transformation requires explicit column aliases`